lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -218,6 +218,12 @@ namespace Js
     {
         Assert(!this->isDetached);
 
+        if (this->IsProjectionArrayBuffer())
+        {
+            Recycler* recycler = GetType()->GetLibrary()->GetRecycler();
```

Comment:
why only projection array buffer?

Reply:
Right. It is logical to report free on any Detach, but that was causing failures and we suspected a separate bug, possibly in a test/library.
Now it looks like we just have the same issue in `JavascriptArrayBuffer` as the one that was fixed for projection buffers - it does not report taking posession of external buffers, but does report free when finalized. If that is fixed, we might not need to specialcase projection buffers.

Reply:
Well, it looks like `WebAssemblyArrayBuffer` also needs fixing. When it grows, it charges for the  _diff_ in sizes, which noone reports freeing later.